core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
Yeah, it makes sense. The only additional question I had was if there's a
way to auto-discover the number of GPUs on the machine and do some math on
it inside the config. i.e. does the config allow bash syntax or something
so I can inspect `/dev/` and count the number of GPUs installed?  I'm
picturing a scenario where I'd like to deploy the same config to a bunch of
hosts, but have the autodiscover this value themselves.

On Thu, Sep 8, 2016 at 3:17 PM Timothy Chen <notifications@github.com>
wrote:

> In
> core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala
> <https://github.com/apache/spark/pull/14644#discussion_r78002761>:
>
> > @@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
> >    private val stateLock = new ReentrantLock
> >
> >    val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
> > +  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
>
> My thoughts was that by only allowing a Boolean flag a spark job either
> uses all GPUs from a host or not, which it won't be able to have different
> GPu devices shared by different jobs. By specifying a limit at least there
> is ability to let a job specify how much GPUs it should grab per node.
> thoughts?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/spark/pull/14644/files/2f658b40251ff75ab4695e4afc55487a6a6eb015#r78002761>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAF4o5c68RBufSO5UF2DpuLtcbkspxClks5qoArjgaJpZM4JkJdn>
> .
>